src/rpncalculator/calc.cpp

```diff
@@ -0,0 +1,253 @@
+                }
+                shuffleDown();
+                break;
+            case keycodes::ke:
+                shuffleUp();
+                stack[0]->setValue(2.718281828459045);
+                break;
+            case keycodes::kpi:
+                shuffleUp();
+                stack[0]->setValue(3.141592653589793);
+                break;
+            case keycodes::kpercent:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(stack[0]->getValue()/100.0);
+                break;
+            case keycodes::kround:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(std::floor(stack[0]->getValue() + 0.5));
+                break;
+            case keycodes::ksqrt:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(std::sqrt(stack[0]->getValue()));
+                break;
+            case keycodes::klog:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(std::log10(stack[0]->getValue()));
+                break;
+            case keycodes::kln:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(std::log(stack[0]->getValue()));
+                break;
+            case keycodes::ksquare:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                res = stack[0]->getValue();
+                stack[0]->setValue(res * res);
+                break;
+            case keycodes::kreciprocal:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(1.0 / stack[0]->getValue());
+                break;
+            case keycodes::kfact:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                res = 1.0;
+                for(auto i = 1.0; i <= stack[0]->getValue(); ++i) {
+                    res *= i;
+                }
+                stack[0]->setValue(res);
+                break;
+            case keycodes::kabs:
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(std::abs(stack[0]->getValue()));
+                break;
+            case keycodes::kpower:
+                res = std::pow(stack[1]->getValue(), stack[0]->getValue());
+                shuffleDown();
+                stack[0]->setValue(res);
+                break;
+            case keycodes::kcosh:
+                res = std::cosh(stack[0]->getValue());
+                if (stack[0]->isBlank()) {
+                    return;
+                }
+                stack[0]->setValue(res);
+                break;
+            case keycodes::ksinh:
```

Comment:
i think i caused a crash by running sinh on number > 1

Reply:
Told you it was buggy :)